### ISSUE TITLE
incompatible with webpack 4.19.1

### ISSUE THREAD

**Issue by dreyks** [mzgoddard/hard-source-webpack-plugin#443]
#442 suffers from the same issue but it's not from terser plugin, it's webpack

from webpack 4.19.1 changelog: `Comments near export default are preserved`

basically all default exports are borked with hard-source

```js
export default anything
```

becomes
```
 __webpack_exports__[\"default\"] = (undefinedanything)
```

reproduction repo https://github.com/dreyks/hswp_fail

run `yarn` and then run `yarn webpack` twice

`dist/main.js`
will contain

```js
/* harmony default export */ __webpack_exports__[\"default\"] = (undefined'wtf')
```

**Comment by chronweigle** [comment]
Hey man, I've had this issue all day. Thanks for finding the problem.

**Comment by OscarBarrett** [comment]
Related issue thread over at webpack: https://github.com/webpack/webpack/issues/8052

**Comment by fakhrishahab** [comment]
Still have no solution about this?

**Comment by ZipoKing** [comment]
Looks like @mzgoddard is not active in here - last commit to master was almost 2 months ago

**Comment by dreyks** [comment]
@anton164 you'll be getting this error with any webpack > 4.19.1 until this issue is fixed

**Comment by hsz** [comment]
@dreyks `>= 4.19.1`

**Comment by ambrt** [comment]
I'm trying hard source on almost vanila Nuxt and i'm getting 
```
× error server-bundle.js:145
  /* harmony default export */ __webpack_exports__["default"] = (undefined{
  ^

  SyntaxError: Unexpected token {
  at new Script (vm.js:79:7)
```

Is this same issue?

**Comment by ZipoKing** [comment]
@ambrt Yes

**Comment by mciszczon** [comment]
I believe I'm getting the same issue here:

```
ERROR in chunk main [entry]
main.2f4a97e125e6bc802243.js
/some/path/node_modules/babel-loader/lib/index.js!/some/path/app/app.js 9f2612c56d09a351475f4d756bbb8f18
Unexpected token (8:91)
| });
| 
| /* harmony default export */ var __WEBPACK_MODULE_DEFAULT_EXPORT__ = (undefinedfunction () {
|   var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : routeInitialState;
|   var action = arguments[1];
```

**Comment by tonyanziano** [comment]
Thank you! We were running into this issue for the past few days. Downgrading webpack to `v4.16.4` fixed the issue for us. 👍

**Comment by dtothefp** [comment]
also getting this error with `4.19.1` and above. Downgrading to `4.19.0` works for me

```js
/* harmony default export */ __webpack_exports__["default"] = (undefinedfunction (routes, pathname) {
```

**Comment by KamalAman** [comment]
Downgrading from 4.23.1 to 4.19.0 resolve this issue for me.

**Comment by Heartz66** [comment]
Just upgraded to webpack 4.25.1, seems to be fixed for me now.

**Comment by ZipoKing** [comment]
Looks like

**Comment by Naeemo** [comment]
You can do this for now:
```
yarn add -D https://github.com/chronweigle/hard-source-webpack-plugin/#add-prefix
```
Thanks to @chronweigle.

**Comment by alexgorbatchev** [comment]
Can confirm that https://github.com/nomcopter/hard-source-webpack-plugin works with webpack@4.25.1

**Comment by krzysztof-kabala** [comment]
removing
```
new HardSourceWebpackPlugin(),
```
and upgrade webpack to @4.26.1
works fine :) 

"hard-source-webpack-plugin" slows down latest webpack only..

**Comment by HostenMarijn** [comment]
> You can do this for now:
> 
> ```
> yarn add -D https://github.com/chronweigle/hard-source-webpack-plugin/#add-prefix
> ```
> Thanks to @chronweigle.

Works for me with webpack 4.25.1

**Comment by mzgoddard** [comment]
Fixed in #477 with @chronweigle's change in #444.
